lib/main.dart

```diff
@@ -1,7 +1,63 @@
+import 'package:provider/provider.dart';
+
+import '../screens/auth_screen.dart';
+import 'package:flutter/material.dart';
+import './screens/profile_screen.dart';
+import './providers/auth.dart';
+import './models/profile.dart';
+import './providers/profiles.dart';
+
 void main() {
   /**
    *    Welcome to Basic authentication challenge
    *    Make sure to read CHALLENGE.md before you start
    *
    */
+  runApp(const MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return MultiProvider(
+      providers: [
+        ChangeNotifierProvider(
+          create: (ctx) => Auth(),
+        ),
+        ChangeNotifierProxyProvider<Auth, Profiles>(
+          create: (ctx) => Profiles(null, null, [], null),
```

Comment:
Having a constructor or a function with many positional parameters is not good in terms of readability. It is better to convert them to named parameters.

Reply:
Updated Profiles to receive named arguments